odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -234,6 +234,32 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 dest_record_id=dst_record.id,
             ))
 
+            # company dependent fields referring the merged records might not be set on the table,
+            # we need to update the default values
+            self.env.cr.execute(SQL(
+                """
+                UPDATE ir_default
+                SET json_value =
+                    CASE
+                        WHEN json_value::int IN %(src_record_ids)s
+                        THEN %(dest_record_id)s::varchar
+                        ELSE json_value::varchar
+                    END
+                WHERE
+                    field_id IN (
+                        SELECT id
+                        FROM ir_model_fields
+                        WHERE name = %(field)
+                        AND model = %(model)s
+                    )
```

Comment:
> Why not use UPDATE ... SET ... FROM ir_model_fields?
and we can directly use ir_model_fields.model to avoid joining ir_model

Didn't see the `model` column on `ir_model_fields` sorry